examples/bind-shopping.amp.html

```diff
@@ -0,0 +1,454 @@
+              "sizes": ["S", "M", "L"]
+            },
+            {
+              "id": "8",
+              "name": "00 WHITE",
+              "image": "./shirts/white.jpeg",
+              "sizes": []
+            }
+          ]
+        }
+      </script>
+    </amp-bind-dataset>
+
+    <!--
+    <amp-carousel slide=0 id="carousel-1" width=1 height=1 layout=responsive type=slides controls
+        bind-attr="{slide: selected}"
+        on="slidechange:setState({selected: ampEventData})">
+    -->
+    <amp-carousel slide=0 id="carousel-1" width=1 height=1 layout=responsive type=slides controls>
+      <amp-bind-attribute layout=nodisplay
+          attr="slide"
+          value="{{selected}}">
+      </amp-bind-attribute>
+      <amp-bind-event layout=nodisplay
+          on="slidechange"
+          variable="selected"
+          value="{{ampEventData}}">
+      </amp-bind-event>
+
+      <amp-img src="./shirts/black.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/brown.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/gray.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/navy.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/wine.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/blue.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/dark-green.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/light-gray.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/white.jpeg" width=1 height=1 layout=responsive></amp-img>
+    </amp-carousel>
+
+    <!--
+    <amp-img class="color" width=355 height=50 src="https://placehold.it/355x50?text=COLOR%20???"
+        bind-src="https://placehold.it/300x400?text=COLOR%20{{items[selected].name}}">
+    -->
+    <amp-img class="color" width=355 height=50 src="https://placehold.it/355x50?text=COLOR%20???">
+      <amp-bind-attribute layout=nodisplay
+          attr="src"
+          value="{{'https://placehold.it/355x50?text=COLOR%20' + items[selected].name}}">
+      </amp-bind-attribute>
+    </amp-img>
+
+    <div class="thumbnails">
+      <!--
+      <amp-img src="./shirts/black.jpeg" width=50 height=50 layout=fixed
+        on="tap:setState({selected: 0})"
+        bind-attr="{selected: selected === 0}"
+        bind-class="{outOfStock: !items[0].sizes.length}">
+      -->
+      <amp-img src="./shirts/black.jpeg" width=50 height=50 layout=fixed>
+        <amp-bind-event layout=nodisplay
+            on="tap"
+            variable="selected"
+            value="{{0}}">
+        </amp-bind-event>
+        <amp-bind-attribute layout=nodisplay
+            attr="selected"
+            toggle="{{selected === 0}}">
+        </amp-bind-attribute>
+        <amp-bind-attribute layout=nodisplay
+            attr="out-of-stock"
```

Comment:
The actual attributes should still be subject of validator. This is not the attribute we allow. Will a class do?